src/main/scala/com/databricks/labs/overwatch/pipeline/Schema.scala

```diff
@@ -283,39 +385,119 @@ object Schema extends SparkSessionWrapper {
+  val minimumScheduleSchema: StructType = StructType(Seq(
+    StructField("pause_status", StringType, true),
+    StructField("quartz_cron_expression", StringType, true),
+    StructField("timezone_id", StringType, true)
+  ))
+
+  val minimumEmailNotificationsSchema: StructType = StructType(Seq(
+    StructField("no_alert_for_skipped_runs", BooleanType, nullable = true),
+    StructField("on_failure", ArrayType(StringType, containsNull = true), nullable = true)
+  ))
+
+  val minimumGitSourceSchema: StructType = StructType(Seq(
+    StructField("git_branch",StringType, nullable = true),
+    StructField("git_provider",StringType, nullable = true),
+    StructField("git_url",StringType, nullable = true),
+    StructField("git_tag",StringType, nullable = true),
+    StructField("git_commit",StringType, nullable = true)
+  ))
+
+  val minimumAccessControlListSchema: ArrayType = ArrayType(
+      StructType(Seq(
+        StructField("user_name",StringType, nullable = true),
+        StructField("group_name",StringType, nullable = true),
+        StructField("permission_level",StringType, nullable = true),
+      ))
+    , containsNull = true)
+
+  val minimumGrantsSchema: ArrayType = ArrayType(
+    StructType(Seq(
+      StructField("permission",StringType, nullable = true),
+      StructField("user_id",LongType, nullable = true)
+    ))
+    ,containsNull = true)
+
+  val minimumJobClustersSchema: ArrayType = ArrayType(
+    StructType(Seq(StructField("job_cluster_key",StringType,nullable = true))),
+    containsNull = true
+  )
+
+  val minimumTasksSchema: ArrayType = ArrayType(StructType(Seq(
+    StructField("task_key",StringType, nullable = true),
+  )))
+
   // minimum new jobs settings struct
   val minimumNewSettingsSchema: StructType = StructType(Seq(
-    StructField("existing_cluster_id", StringType, true),
-    StructField("max_concurrent_runs", LongType, true),
-    StructField("name", StringType, true),
-    StructField("new_cluster", minimumNewClusterSchema, true),
-    StructField("timeout_seconds", LongType, true)
+    StructField("existing_cluster_id", StringType, nullable = true),
+    StructField("max_concurrent_runs", LongType, nullable = true),
+    StructField("name", StringType, nullable = true),
+    StructField("new_cluster", minimumNewClusterSchema, nullable = true),
+    StructField("timeout_seconds", LongType, nullable = true),
+    StructField("notebook_task", minimumNotebookTaskSchema, nullable = true),
+    StructField("spark_python_task", minimumSparkPythonTaskSchema, nullable = true),
+    StructField("python_wheel_task", minimumPythonWheelTaskSchema, nullable = true),
+    StructField("spark_jar_task", minimumSparkJarTaskSchema, nullable = true),
+    StructField("spark_submit_task", minimumSparkSubmitTaskSchema, nullable = true),
+    StructField("shell_command_task", minimumShellCommandTaskSchema, nullable = true),
+    StructField("pipeline_task", minimumPipelineTaskSchema, nullable = true),
```

Comment:
add `sql_task` and `dbt_task`

Reply:
same as above, not selected anywhere; thus they don't need to be here -- if they become a downstream used filed, they will be added.

Reply:
actually -- this entire object should be deleted as it's no longer used. We converted golds to no longer reference the raw, `new_settings` field but instead hand pick the values we want. Also jobs team is changing the task details publication logic soon